tests/test_clean.py

```diff
@@ -305,11 +505,44 @@ def test_attributes_list():
 
     text = u'<a href="/foo" title="blah">example</a>'
     assert (
-        bleach.clean(text, tags=TAGS, attributes=ATTRS) ==
+        clean(text, tags=TAGS, attributes=ATTRS) ==
         u'<a title="blah">example</a>'
     )
 
 
+@pytest.mark.parametrize('data, kwargs, expected', [
+    # javascript: is not allowed by default
+    (
+        '<a href="javascript:alert(\'XSS\')">xss</a>',
+        {},
+        '<a>xss</a>'
+    ),
+
+    # File protocol is not allowed by default
+    (
+        '<a href="file:///tmp/foo">foo</a>',
+        {},
+        '<a>foo</a>'
+    ),
+
+    # Specified protocols are allowed
+    (
+        '<a href="myprotocol://more_text">allowed href</a>',
+        {'protocols': ['myprotocol']},
+        '<a href="myprotocol://more_text">allowed href</a>'
+    ),
+
+    # Unspecified protocols are not allowed
+    (
+        '<a href="http://xx.com">invalid href</a>',
+        {'protocols': ['myprotocol']},
+        '<a>invalid href</a>'
+    )
+])
+def test_uri_value_allowed_protocols(data, kwargs, expected):
+    assert clean(data, **kwargs) == expected
```

Comment:
These tests were kind of all over the place, so I merged them into a single parametrized test.